nncf/quantization/algorithms/channel_alignment/openvino_backend.py

```diff
@@ -0,0 +1,162 @@
+from nncf.quantization.algorithms.channel_alignment.backend import LayoutDescriptor
+
+
+@ALGO_BACKENDS.register(BackendType.OPENVINO)
+class OVChannelAlignmentAlgoBackend(ChannelAlignmentAlgoBackend):
+    @staticmethod
+    def target_point(target_type: TargetType, target_node_name: str, port_id: int) -> OVTargetPoint:
+        return OVTargetPoint(target_type, target_node_name, port_id)
+
+    @staticmethod
+    def get_bias_value(node: NNCFNode, model: ov.Model, nncf_graph: NNCFGraph) -> np.ndarray:
+        return get_bias_value(node, nncf_graph, model)
+
+    @staticmethod
+    def get_weight_value(node: NNCFNode, model: ov.Model, port_id: int) -> np.ndarray:
+        return get_weight_value(node, model, port_id)
+
+    @staticmethod
+    def get_activation_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        return 0, 0
+
+    @staticmethod
+    def get_weights_port_ids_for_node(node: NNCFNode) -> Tuple[int, int]:
+        return 0, 1
+
+    @staticmethod
+    def get_conv_metatypes():
+        return [OVConvolutionMetatype, OVGroupConvolutionMetatype, OVDepthwiseConvolutionMetatype]
+
+    @staticmethod
+    def get_linear_metatypes():
+        return [OVMatMulMetatype]
+
+    @staticmethod
+    def get_add_metatypes():
+        return [OVAddMetatype, OVSubtractMetatype]
+
+    @staticmethod
+    def get_statistic_collector(
+        reduction_shape, q: float, num_samples: int, inplace: bool
+    ) -> TensorStatisticCollectorBase:
+        tensor_collector = TensorCollector(OVMinMaxTensorStatistic)
+        quantile_reducer = OVQuantileReducer(reduction_shape, (q, 1 - q), inplace)
+
+        for port_id, container_key in enumerate([OVMinMaxTensorStatistic.MIN_STAT, OVMinMaxTensorStatistic.MAX_STAT]):
+            aggregator = MedianAggregator(OVNNCFCollectorTensorProcessor, num_samples=num_samples)
+            tensor_collector.register_statistic_branch(container_key, quantile_reducer, aggregator, port_id)
+        return tensor_collector
+
+    @staticmethod
+    def is_node_with_bias(node: NNCFNode, nncf_graph: NNCFGraph) -> bool:
+        next_nodes = nncf_graph.get_next_nodes(node)
+        if not next_nodes:
+            return False
+
+        add_node = next_nodes[0]
+        if add_node.metatype != OVAddMetatype:
+            return False
+
+        bias_constant = get_node_with_bias_value(add_node, nncf_graph)
+        return bias_constant is not None
+
+    @staticmethod
+    def create_bias_update_command(
+        node_with_bias: NNCFNode, updated_value: np.ndarray, nncf_graph: NNCFGraph
+    ) -> OVWeightUpdateCommand:
+        return OVCommandCreator.create_command_to_update_bias(node_with_bias, updated_value, nncf_graph)
+
+    @staticmethod
+    def create_weights_update_command(
+        node_with_weights: NNCFNode, updated_value: np.array, weights_port_id: int
+    ) -> OVWeightUpdateCommand:
+        return OVCommandCreator.create_command_to_update_weight(node_with_weights, updated_value, weights_port_id)
```

Comment:
You can create commands inside common code using `CommandCreator` and `CommandCreatorFactory`

Reply:
Done!